src/util.rs

```diff
@@ -0,0 +1,140 @@
+// -*- coding: utf-8 -*-
```

Comment:
I would suggest a different filename — `util` is very generic.  Maybe `parse_error.rs`?

Reply:
That is a good idea.